backend/src/main/java/de/bund/digitalservice/ris/norms/application/validator/Validator.java

```diff
@@ -0,0 +1,15 @@
+package de.bund.digitalservice.ris.norms.application.validator;
+
+import de.bund.digitalservice.ris.norms.utils.exceptions.ValidationException;
+
+/** Interface to be implemented by any custom validator. */
+public interface Validator {
```

Comment:
We could also use the existing interface for validators from [spring](https://docs.spring.io/spring-framework/docs/6.1.9/javadoc-api/org/springframework/validation/Validator.html) or [jakarta](https://jakarta.ee/specifications/platform/9/apidocs/jakarta/validation/validator). That would "for free" provide us with some convenient ways of calling the validator like using the `@Valid` annotation and the option to use more complex validation errors.

Reply:
Oh i didn't know about that. But then we could need to create custom classes for the input parameters passed to the validator right? Like the `SingleModValidator` would need a class containing a Norm and also a Mod. I think this validator interface is a good fit for domain entities that you already have. Or you think it is worth the time investigating a bit further?

Reply:
I agree that at the moment this `Validator` interface would work totally fine.
But if we ever feel the need to extend it further / support more things I think we should investigate using an existing solution. I would expect that there are a lot of things we haven't even thought of that they have already solved for us as validation is a pretty common problem.

Reply:
I understand. Let's keep this in mind for when we extend our validation process